packages/SwingSet/src/kernel/kernel.js

```diff
@@ -936,11 +936,22 @@ export default function buildKernel(
       return results;
     }
 
-    const args = [upgradeID, true, undefined, incarnationNumber];
-    const vatAdminMethargs = {
-      body: JSON.stringify(['vatUpgradeCallback', args]),
-      slots: [],
-    };
+    // TODO: Use a real makeMarshal serialize?
```

Comment:
Same question. The sooner we get rid of non-local knowledge of serialization formats, the better. It would also make this code easier to understand and review.